permission_handler/lib/permission_handler.dart

```diff
@@ -33,7 +33,7 @@ extension PermissionActions on Permission {
   ///
   /// This is only implemented on Android, calling this on iOS always returns
   /// [false].
-  Future<bool> get shouldShowRequestRationale async {
+  Future<bool?> get shouldShowRequestRationale async {
```

Comment:
This should also not be allowed to return a `null` value:

- `true` if Android indicates the permission rationale should be shown;
- `false` if Android indicates if the permission rationale should not be shown or if the platform is NOT Android;
- Or an exception if something went wrong.